### PR TITLE
[Build, Compiler, Refactor] Upgrade typescript and add optional-chaining and nullish-coalescing

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary

This PR introduces JS/TS features [Optional Chaining](https://github.com/tc39/proposal-optional-chaining) (`?.`) and [Nullish Coalescing](https://github.com/tc39/proposal-nullish-coalescing) (`??`). I believe the timing is appropriate with the recent alignment between TypeScript 3.7.x and TC39 proposals reaching stages 4 and 3.

* Upgrades TypeScript to `3.7.4`, stable/"final" since `3.7.2`. (This handles the `.ts/.tsx` side of things.)
* Adds Babel plugins `@babel/plugin-proposal-optional-chaining` and `@babel/plugin-proposal-nullish-coalescing-operator`. (This handles the `.js/.jsx` side of things.)
* Adds `coverage` folder to `.gitignore` to make working with `npm run test:coverage` easier.

Also included is an introductory refactoring to show how these new operators could be helpful. (See changes to `utils/youtube.js`.) 

NOTE

If your editor (e.g. VS Code) is not recognizing the newer syntax, you may need to delete `node_modules` and/or re-select your TypeScript version.

#### Ticket Link
None

#### Related Pull Requests
None

#### Screenshots
N/A, no UI/UX changes.

### CODE REVIEW

File: utils/youtube.js [hunk 1]
```diff
@@ -4,32 +4,32 @@
...
 
-    const hours = time[2] ? time[2].match(/([0-9]+)h/) : null;
-    const minutes = time[3] ? time[3].match(/([0-9]+)m/) : null;
-    const seconds = time[4] ? time[4].match(/([0-9]+)s?/) : null;
+    const hours = time[2]?.match(/([0-9]+)h/) ?? null;
+    const minutes = time[3]?.match(/([0-9]+)m/) ?? null;
+    const seconds = time[4]?.match(/([0-9]+)s?/) ?? null;
 
-    let ticks = 0;
+    let startSeconds = 0;
```
Comment (calebroseland) [inline utils/youtube.js]: renamed for clarity as `&start=` is seconds here

File: package.json [hunk 2]
```diff
@@ -65,15 +65,17 @@
...
     "xregexp": "4.2.4",
     "zen-observable": "0.8.14"
   },
   "devDependencies": {
     "@babel/cli": "7.6.4",
     "@babel/core": "7.6.4",
     "@babel/plugin-proposal-class-properties": "7.5.5",
+    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.7.4",
     "@babel/plugin-proposal-object-rest-spread": "7.6.2",
+    "@babel/plugin-proposal-optional-chaining": "^7.7.5",
```
Comment (saturninoabril) [inline package.json]: Please use fix versions
```suggestion
    "@babel/plugin-proposal-optional-chaining": "7.7.5",
```

Comment (calebroseland) [reply package.json]: Thanks for catching that! And for mentioning the codemod—I wasn't aware of that one.

**Comment by hanzei** [comment]
@wiersgallak @jaydeland Would you please review this PR soon? It should be great if developers could use this feature soon.

**Comment by jaydeland** [comment]
> @wiersgallak @jaydeland Would you please review this PR soon? It should be great if developers could use this feature soon.

@hanzei - I am not sure what value I add to this review from a DevOps/Infra standpoint.

**Comment by hmhealey** [comment]
Re-assigning the second dev review to @jespino since he's more familiar with the TypeScript migration.

**Pull request merged**